rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/expression/ExpressionGenerator.xtend

```diff
@@ -1176,4 +1178,63 @@ class ExpressionGenerator extends RosettaExpressionSwitch<JavaStatementBuilder,
 	override protected caseToZonedDateTimeOperation(ToZonedDateTimeOperation expr, Context context) {
 		conversionOperation(expr, context, '''«ZonedDateTime»::parse''', DateTimeParseException)
 	}
+
+	override protected caseSwitchOperation(SwitchOperation expr, Context context) {
+ 		val switchArgument = expr.argument.javaCode(MAPPER.wrap(typeProvider.getRType(expr.argument).toJavaReferenceType), context.scope)
+ 		val caseStatements = expr.values
+ 		val defaultExpression = expr.^default
+
+
+ 		val conditionType = switchArgument.expressionType.itemType
+
+ 		val returnType = MAPPER.wrap(join(
+ 				caseStatements.map[typeProvider.getRType(it.expression)] + (defaultExpression === null
+ 					? #[]
+ 					: #[
+ 					typeProvider.getRType(defaultExpression)
+ 				]
+ 			)
+ 		).toJavaReferenceType)
+
+ 		switchArgument
+ 			.declareAsVariable(true, "switchAgument", context.scope)
+ 			.mapExpression[
+ 				createSwitchJavaExpression(conditionType, returnType, it, caseStatements, defaultExpression, context.scope)
+ 			]
+ 	}
+
+ 	private def JavaStatementBuilder createSwitchJavaExpression(JavaType conditionType, JavaType returnType, JavaExpression switchArgument,
+ 		CaseStatement[] caseStatements, RosettaExpression defaultExpression, JavaScope javaScope) {
+ 		val head = caseStatements.head
+ 		val tail = caseStatements.tail
+
+ 		val javaStatment = if (head.literalCondition !== null) {
```

Comment:
Potential null pointer: for a switch statement with no cases (potentially only a default) -- `head` is null.

Reply:
Also: typo in `javaStatment`?